src/com/seafile/seadroid2/ui/activity/SearchActivity.java

```diff
@@ -0,0 +1,423 @@
+        }
+    }
+
+    public void onSearchedFileSelected(SearchedFile searchedFile) {
+        final String repoID = searchedFile.getRepoID();
+        SeafRepo seafRepo = dataManager.getCachedRepoByID(repoID);
+        final String repoName = seafRepo.getName();
+        final String filePath = searchedFile.getPath();
+
+        if (searchedFile.isDir()) {
+            if (seafRepo == null) {
+                showToast("Couldn't find this library. It may be deleted");
+                return;
+            }
+            showRepo(repoID, repoName, filePath, null);
+            //showToast(R.string.search_open_dir_not_support);
+            return;
+        }
+
+        File localFile = dataManager.getLocalCachedFile(repoName, repoID, filePath, null);
+        if (localFile != null) {
+            showFile(localFile);
+            return;
+        }
+
+        startFileActivity(repoName, repoID, filePath);
+    }
+
+    private void startFileActivity(String repoName, String repoID, String filePath) {
+        int taskID = txService.addDownloadTask(account, repoName, repoID, filePath);
+        Intent intent = new Intent(this, FileActivity.class);
+        intent.putExtra("repoName", repoName);
+        intent.putExtra("repoID", repoID);
+        intent.putExtra("filePath", filePath);
+        intent.putExtra("account", account);
+        intent.putExtra("taskID", taskID);
+        startActivity(intent);
+    }
+
+    private void showRepo(String repoID, String repoName, String path, String dirID) {
+        Intent intent = new Intent(this, BrowserActivity.class);
+        intent.putExtra("repoID", repoID);
+        intent.putExtra("repoName", repoName);
+        intent.putExtra("path", path);
+        intent.putExtra("dirID", dirID);
+        startActivity(intent);
+    }
+
+    public void showFile(File file) {
+        String name = file.getName();
+        String suffix = name.substring(name.lastIndexOf('.') + 1).toLowerCase();
+
+        if (suffix.length() == 0) {
+            showToast(R.string.unknown_file_type);
+            return;
+        }
+
+        if (suffix.endsWith("md") || suffix.endsWith("markdown")) {
+            startMarkdownActivity(file.getPath());
+            return;
+        }
+
+        String mime = MimeTypeMap.getSingleton().getMimeTypeFromExtension(suffix);
+        Intent open = new Intent(Intent.ACTION_VIEW);
+        open.setDataAndType((Uri.fromFile(file)), mime);
+
+        try {
+            startActivity(open);
+            return;
+        } catch (ActivityNotFoundException e) {
+            new OpenAsDialog(file).show(getSupportFragmentManager(), "OpenAsDialog");
+            //showToast(R.string.activity_not_found);
+            return;
+        }
+
+    }
+
+    private void startMarkdownActivity(String path) {
+        Intent intent = new Intent(this, MarkdownActivity.class);
+        intent.putExtra("path", path);
+        startActivity(intent);
+    }
+
+    public void showToast(int id) {
```

Comment:
Lots of our classes has implemented `showToast` method repeatedly, we should refactor them out, like https://github.com/github/android/blob/9d490829b944d3a2c77dbd0010ec7a0bfe2efaee/app/src/main/java/com/github/mobile/util/ToastUtils.java#L39.